### PR TITLE
softmaker-office-nx: init at 1222

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Things done

Add [SoftMaker Office NX](https://www.softmaker.com/fr/assistance/installation/linux/office-nx), a proprietary office suite.

- **softmaker-office-nx: init at 1222**
- **softmaker-office: rename source file**

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 366490`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>softmaker-office-nx</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -14958,7 +14958,9 @@ with pkgs;
 
   smtube = libsForQt5.callPackage ../applications/video/smtube { };
 
-  softmaker-office = callPackage ../applications/office/softmaker/softmaker_office.nix { };
+  softmaker-office-nx = callPackage ../applications/office/softmaker/softmaker-office-nx.nix { };
```
Comment (kirillrdy) [inline pkgs/top-level/all-packages.nix:14961]:
```
 Attribute `pkgs.softmaker-office-nx` is a new top-level package using `pkgs.callPackage ./pkgs/applications/office/softmaker/softmaker-office-nx.nix { /* ... */ }`.
  Please define it in pkgs/by-name/so/softmaker-office-nx/package.nix instead.
```

Comment (GaetanLepage) [reply pkgs/top-level/all-packages.nix:14961]: Yes, I asked in matrix. I'm not sure what to do here.
As the derivation depends on the shared `generic.nix`, it seems weird to split everything apart.
But maybe I missed a smarter design.

**Comment by liberodark** [comment]
Hi,

If this PR is merge i think this PR can be close : 
https://github.com/NixOS/nixpkgs/pull/366476
One use generic.nix other not.

Best Regars.

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (kirillrdy) [reply pkgs/top-level/all-packages.nix:14961]: there is a workaround that is something like `inherit ( import genetic.nix ) foo1 foo2 `

Comment (kirillrdy) [reply pkgs/top-level/all-packages.nix:14961]: sorry i meant 

```
inherit (callPackage ../applications/office/softmaker/softmaker-office-nx.nix { })  softmaker-office-nx ;
```

Comment (infinisil) [reply pkgs/top-level/all-packages.nix:14961]: Check out https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name#recommendation-for-new-packages-with-multiple-versions

**Review by kirillrdy — APPROVED** [review]

**Pull request merged**